packages/now-next/src/index.ts

```diff
@@ -901,17 +887,28 @@ export const build = async ({
       ...staticDirectoryFiles,
     },
     routes: [
-      ...topRoutes,
+      // Before we handle static files we need to set proper caching headers
+      {
+        // This ensures we only match known emitted-by-Next.js files and not
+        // user-emitted files which may be missing a hash in their filename.
+        src: path.join(
+          '/',
+          entryDirectory,
+          '_next/static/(?:[^/]+/pages|chunks|runtime|css|media)/.+'
+        ),
+        // Next.js assets contain a hash or entropy in their filenames, so they
+        // are guaranteed to be unique and cacheable indefinitely.
+        headers: { 'cache-control': 'public,max-age=31536000,immutable' },
+        continue: true,
+      },
+      { src: path.join('/', entryDirectory, '_next(?!/data(?:/|$))(?:/.*)?') },
+      // Next.js page lambdas, `static/` folder, reserved assets, and `public/`
+      // folder
+      { handle: 'filesystem' },
       ...redirects,
       ...rewrites,
-      // we need to re-apply the routes above rewrites in-case the are
-      // rewriting to one of those routes
-      ...topRoutes,
       // Static exported pages (.html rewrites)
       ...exportedPageRoutes,
```

Comment:
We need to remove these routes, then set the contentType and remove the extension on the file output instead

Reply:
Related: #3347